ping_manager/ping_manager.py

```diff
@@ -0,0 +1,247 @@
+    
+    @commands.group(invoke_without_command=True)
+    @checks.has_permissions(PermissionLevel.REGULAR)
+    async def ping_ignore_categories(self, ctx: commands.Context) -> None:
+        """Manage what categories never get sent pings in them."""
+        await ctx.send_help(ctx.command)
+    
+    @ping_ignore_categories.command(name="add")
+    async def set_ping(self, ctx: commands.Context, category_to_ignore: discord.CategoryChannel) -> None:
+        """Add a category to the list of ignored categories."""
+        await self.init_task
+
+        if category_to_ignore.id in self.config.ignored_categories:
+            await ctx.send(f":x: {category_to_ignore} already in the ignored categories.")
+            return
+
+        self.config.ignored_categories.append(category_to_ignore.id)
+        await self.db.find_one_and_update(
+            {"_id": "ping-delay-config"},
+            {"$set": {"ignored_categories": self.config.ignored_categories}},
+            upsert=True,
+        )
+        await ctx.send(f":+1: Added {category_to_ignore} to the ignored categories list.")
+
+    @ping_ignore_categories.command(name="get")
+    async def get_ping(self, ctx: commands.Context) -> None:
+        """Get the list of ignored categories."""
+        await self.init_task
+
+        if not self.config.ignored_categories:
+            await ctx.send(f"There are currently not ignored categories.")
+            return
+
+        ignored_categories_str = ', '.join(map(str, self.config.ignored_categories))
+        await ctx.send(f"The currently ignored categories are: {ignored_categories_str}.")
+    
+    @ping_ignore_categories.command(name="delete", aliases=("remove", "del"))
+    async def del_ping(self, ctx: commands.Context, category_to_ignore: discord.CategoryChannel) -> None:
+        """Remove a category from the list of ignored categories."""
+        await self.init_task
+
+        if category_to_ignore.id not in self.config.ignored_categories:
+            await ctx.send(f":x: {category_to_ignore} isn't in the ignored categories list.")
+            return
+
+        self.config.ignored_categories.remove(category_to_ignore.id)
+        await self.db.find_one_and_update(
+            {"_id": "ping-delay-config"},
+            {"$set": {"ignored_categories": self.config.ignored_categories}},
+            upsert=True,
+        )
+        await ctx.send(f":+1: Removed {category_to_ignore} from the ignored categories list.")
+
+    async def should_ping(self, channel: discord.TextChannel) -> bool:
+        """Check if a ping should be sent to a thread depending on current config."""
+        if channel.category_id in self.config.ignored_categories:
+            log.info("Not pinging in %s as it's currently in an ignored category", channel)
+            return False
+        
+        first_message_list = await channel.history(limit=1, oldest_first=True).flatten()
```

Comment:
Will need to catch NotFound errors here since the channel may already be closed.

Reply:
Resolved in https://github.com/python-discord/modmail-plugins/pull/4/commits/23ea53e57ad9418d6a848b05287705b6df5123a1